packages/terra-alert/src/Alert.jsx

```diff
@@ -0,0 +1,218 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import ResponsiveElement from 'terra-responsive-element';
+import Button from 'terra-button';
+import IconAlert from 'terra-icon/lib/icon/IconAlert';
+import IconError from 'terra-icon/lib/icon/IconError';
+import IconWarning from 'terra-icon/lib/icon/IconWarning';
+import IconRequired from 'terra-icon/lib/icon/IconRequired';
+import IconDiamond from 'terra-icon/lib/icon/IconDiamond';
+import IconInformation from 'terra-icon/lib/icon/IconInformation';
+import IconSuccess from 'terra-icon/lib/icon/IconSuccess';
+import classNames from 'classnames/bind';
+import 'terra-base/lib/baseStyles';
+import styles from './Alert.scss';
+
+const cx = classNames.bind(styles);
+
+const AlertTypes = {
+  ALERT: 'alert',
+  ERROR: 'error',
+  WARNING: 'warning',
+  ADVISORY: 'advisory',
+  INFO: 'info',
+  SUCCESS: 'success',
+  GAP_CHECKING: 'gapChecking',
+  OUTSIDE_RECORDS: 'outsideRecords',
+  CUSTOM: 'custom',
+};
+
+const propTypes = {
+  /**
+   * An action element to be added to the action section of the alert.
+   */
+  alertAction: PropTypes.element,
+  /**
+   * Child Nodes providing the message content for the alert. Can contain text and HTML.
+   */
+  children: PropTypes.node,
+  /**
+   * The icon to be used for an alert of type custom.
+   */
+  customIcon: PropTypes.element,
+  /**
+   * The status bar color to be used for an alert of type custom.
+   */
+  customStatusColor: PropTypes.string,
+  /**
+   * Callback function triggered when Dismiss button is clicked. The presence of this prop will cause the Dismiss button to be included on the alert.
+   */
+  onDismiss: PropTypes.func,
+  /**
+   * The title for the alert which will be bolded.
+   */
+  title: PropTypes.string,
+  /**
+   * The type of alert to be rendered. One of Alert.Opts.Types.ALERT, Alert.Opts.Types.ERROR, Alert.Opts.Types.WARNING, Alert.Opts.Types.ADVISORY,
+   * Alert.Opts.Types.INFO, Alert.Opts.Types.SUCCESS, Alert.Opts.Types.GAP_CHECKING, Alert.Opts.Types.OUTSIDE_RECORDS, Alert.Opts.Types.CUSTOM.
+   * Use the Alert.Opts.Types attribute of the Alert component for access to these type strings.
+   */
+  type: PropTypes.oneOf([
+    AlertTypes.ALERT,
+    AlertTypes.ERROR,
+    AlertTypes.WARNING,
+    AlertTypes.ADVISORY,
+    AlertTypes.INFO,
+    AlertTypes.SUCCESS,
+    AlertTypes.GAP_CHECKING,
+    AlertTypes.OUTSIDE_RECORDS,
+    AlertTypes.CUSTOM,
+  ]),
+};
+
+const defaultProps = {
+  alertAction: null,
+  children: '',
+  customIcon: null,
+  customStatusColor: '',
+  onDismiss: null,
+  title: '',
+  type: AlertTypes.ALERT,
+};
+
+const getAlertIcon = (type, customIcon) => {
```

Comment:
I think the "react way" is to move this logic into a component. E.g. we'd create an `AlertIcon` component and you'd use it like:

```jsx
 <div className={cx('icon')}>
   <AlertIcon type={type} customIcon={customIcon} />
</div>
```

FWIW, this will also make the alert Icon logic easier to test IMO.